packages/rocketchat-statistics/server/lib/SAUMonitor.js

```diff
@@ -0,0 +1,410 @@
+			this._today = currentDay;
+			return;
+		}
+
+		// Otherwise, just update the lastActivityAt field
+		this._applyAllServerSessionsIds((sessions) => {
+			Sessions.updateActiveSessionsByDateAndInstanceIdAndIds({ year, month, day }, this._instanceId, sessions, { lastActivityAt: currentDateTime });
+		});
+	}
+
+	_getConnectionInfo(connection, params = {}) {
+		if (!connection) {
+			return;
+		}
+
+		const ip = connection.httpHeaders ? connection.httpHeaders['x-real-ip'] || connection.httpHeaders['x-forwarded-for'] : connection.clientAddress;
+		const host = connection.httpHeaders && connection.httpHeaders.host;
+		const info = {
+			type: 'session',
+			sessionId: connection.id,
+			instanceId: this._instanceId,
+			ip,
+			host,
+			...this._getUserAgentInfo(connection),
+			...params,
+		};
+
+		if (connection.loginAt) {
+			info.loginAt = connection.loginAt;
+		}
+
+		return info;
+	}
+
+	_getUserAgentInfo(connection) {
+		if (!(connection && connection.httpHeaders && connection.httpHeaders['user-agent'])) {
+			return;
+		}
+
+		const uaString = connection.httpHeaders['user-agent'];
+		let result;
+
+		if (UAParserMobile.isMobileApp(uaString)) {
+			result = UAParserMobile.uaObject(uaString);
+		} else {
+			const ua = new UAParser(uaString);
+			result = ua.getResult();
+		}
+
+		const info = {
+			type: 'other',
+		};
+
+		const removeEmptyProps = (obj) => {
+			Object.keys(obj).forEach((p) => (!obj[p] || obj[p] === undefined) && delete obj[p]);
+			return obj;
+		};
+
+		if (result.browser && result.browser.name) {
+			info.type = 'browser';
+			info.name = result.browser.name;
+			info.longVersion = result.browser.version;
+		}
+
+		if (result.os && result.os.name) {
+			info.os = removeEmptyProps(result.os);
+		}
+
+		if (result.device && (result.device.type || result.device.model)) {
+			info.type = 'mobile-app';
+
+			if (result.app && result.app.name) {
+				info.name = result.app.name;
+				info.longVersion = result.app.version;
+				if (result.app.bundle) {
+					info.longVersion += ` ${ result.app.bundle }`;
+				}
+			}
+		}
+
+		if (typeof info.longVersion === 'string') {
+			info.version = info.longVersion.match(/(\d+\.){0,2}\d+/)[0];
+		}
+
+		return {
+			device: info,
+		};
+	}
+
+	_initActiveServerSessions() {
+		this._applyAllServerSessions((connectionHandle) => {
+			this._handleSession(connectionHandle, getDateObj());
+		});
+	}
+
+	_applyAllServerSessions(callback) {
+		if (!callback || typeof callback !== 'function') {
+			return;
+		}
+
+		const sessions = Object.values(Meteor.server.sessions).filter((session) => session.userId);
+		sessions.forEach((session) => {
+			callback(session.connectionHandle);
+		});
+	}
+
+	_applyAllServerSessionsIds(callback) {
+		if (!callback || typeof callback !== 'function') {
+			return;
+		}
+
+		const sessions = Object.values(Meteor.server.sessions).filter((session) => session.userId);
```

Comment:
```suggestion
		const sessions = Object.values(Meteor.server.sessions).filter((session) => session.userId).map((s) => s.id);
```

`filter` and `map` can be "grouped" so the line below can be removed.